core/primitives-core/src/config.rs

```diff
@@ -68,6 +68,9 @@ pub struct VMLimitConfig {
     pub max_length_returned_data: u64,
     /// Max contract size
     pub max_contract_size: u64,
+    /// Max transaction size
+    #[cfg(feature = "protocol_feature_tx_size_limit")]
+    pub max_transaction_size: u64,
```

Comment:
That's mostly unrelated to the PR, but I am somewhat surprised that we use raw `u64` for quantities, instead of using strongly typed wrappers a-la `pub struct Bytes(pub u64);`, `pub struct Gas(pub u64);`, etc. For byte sizes, we started to use https://github.com/hyunsik/bytesize somewhere in network.

Reply:
I guess `Gas` and `Balance` have been separated into such types, because they are widely used and it makes sense to change type size everywhere, if we want to.
But it's not simple to find cases where other quantities in `VMLimitConfig` can be united into other types.
What is the common sense beneath replacing raw types with strongly typed wrappers?
Also, apparently `VMLimitConfig` is so widely used so it could be simpler to create a separate issue for this.

Reply:
There are big benefits, but a bunch of small ones. 

Units become unambiguous. If you have `size: Bytes`, you don't need to ask whether the unit is byte or page, you can *ask* for what you need: `size.as_bytes()`, `size.as_kilobytes`. You can say `timeout: Duration` rather then `timeout_ms: u64`.

Common functionality becomes available. I see a couple of places that did `/ 1024` for formatting purposes. With a type, you can just write `size.as_kilobytes` or even `format!("{:?}", duration)` and have the type automatically pick a human-readable formatting. 

Reduced probability of type confusion error: it's impossible to add apples and oranges if the two are different types. 

(If we implement this our selves): we can avoid providing unchecked arithmetics. Instead of various `safe_add_stuff` utils which you need to remember to use, there'll be *only* checked methods like `size.add(other_size)?`.

Reply:
Makes sense. Created a separate issue https://github.com/near/nearcore/issues/4139